src/inference/gibbs_conditional.jl

```diff
@@ -0,0 +1,149 @@
+struct GibbsConditional{space, C} <: InferenceAlgorithm 
+    conditional::C
+
+    function GibbsConditional(sym::Symbol, conditional::C) where {C}
+        return new{sym, C}(conditional)
+    end
+end
+
+getspace(::GibbsConditional{S}) where {S} = (S,)
+alg_str(::Sampler{<:GibbsConditional}) = "GibbsConditional"
+
+function Sampler(
+    alg::GibbsConditional,
+    model::Model,
+    s::Selector=Selector()
+)
+    # Set up info dict.
+    info = Dict{Symbol, Any}()
+
+    # Make a varinfo.
+    vi = VarInfo(model)
+
+    # Set up state struct.
+    state = SamplerState(vi)
+
+    # Generate a sampler.
+    spl = Sampler(alg, info, s, state)
+
+    return spl
+end
+
+function AbstractMCMC.sample_init!(
+    rng::AbstractRNG,
+    model::Model,
+    spl::Sampler{<:GibbsConditional},
+    N::Integer;
+    verbose::Bool=true,
+    resume_from=nothing,
+    kwargs...
+)
+    # Resume the sampler.
+    set_resume!(spl; resume_from=resume_from, kwargs...)
+
+    # Get `init_theta`
+    initialize_parameters!(spl; verbose=verbose, kwargs...)
+end
```

Comment:
Actually I think you can remove `sample_init!` as well.
```suggestion
```
I think the default implementation in https://github.com/TuringLang/Turing.jl/blob/bcd5ab99c26ebb111d0accf9be2e51ce725abbc3/src/inference/Inference.jl#L184-L196 is sufficient.